### PR TITLE
changelog for Pro 3.4 and CityEngine 2024.1

### PR DESCRIPTION
**Pull request by matthias-specht**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,3 +1,78 @@
...
+  * cleanupGeometry operation: Removing coplanar edges now preserves the first edge.
+  * Floating point precision:
+      * All internal geometry processing was switched form float to double precision. The numerical behaviour is more accurate now. It is possible that some rules change behaviour because e.g a float value comparison gives a different result now.
+      * Float to string conversion (e.g. str() function and print() operation): fixed wrong string values in rare cases. Further, the maximum number of digits in the string representation was reduced by 1.
+
+## Built-In Codecs
+* ShapeBuffer Encoder:
+   * Disabled vertex / normal / uv merging. This preserves small geometric details and improves performance.
+* USD Decoder:
+   * Fixed a race condition which led to decode failures if multiple assets with the same ResolveMap key were decoded in the same time.
```
Comment (SGoetschi) [inline changelog.md:32]:
```suggestion
   * Fixed a race condition which led to decode failures if multiple assets with the same ResolveMap key were decoded at the same time.  
```

File: changelog.md [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+      * Float to string conversion (e.g. str() function and print() operation): fixed wrong string values in rare cases. Further, the maximum number of digits in the string representation was reduced by 1.
+
+## Built-In Codecs
+* ShapeBuffer Encoder:
+   * Disabled vertex / normal / uv merging. This preserves small geometric details and improves performance.
+* USD Decoder:
+   * Fixed a race condition which led to decode failures if multiple assets with the same ResolveMap key were decoded in the same time.  
+
+## Misc Changes and Fixes
+
```
Comment (SGoetschi) [inline changelog.md:35]:
```suggestion
```

Comment (SGoetschi) [reply changelog.md:35]: Remove the empty section. E.g. in 3.1.9666 we also don't have it.

File: changelog.md [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+# CITYENGINE SDK 3.2.10352 CHANGELOG
+
+This section lists changes compared to CityEngine SDK 3.2.9903.
```
Comment (SGoetschi) [inline changelog.md:38]:
```suggestion
This section lists changes compared to CityEngine SDK 3.2.10211.
```

**Review by SGoetschi — CHANGES_REQUESTED** [review]
Mostly great. Just some minor things.

File: changelog.md [hunk 4]
```diff
@@ -1,3 +1,78 @@
...
+  * Fixed a crash when faces had less than 3 vertices. Such faces are ignored now.
+  * Fixed a bug where the first edge of triangle fans and strips was wrong.
+* OBJ and GLTF Encoders:
+  * Non-ASCII characters are supported now. Previously these were replaced by '_'.
+  * No '_0' suffix is added to the filename anymore if the "memory budget" option is disabled.
+* DAE, FBX and VOB Encoders:
+  * No '_0' suffix is added to the filename anymore if the "memory budget" option is disabled.
+* USD Decoder:
+  * Fixed a bug where duplicate materials were created (instead of one shared material).
+  * Fixed a bug that Material attribute overrides in USD meshes where ignored by the importer in some cases.
```
Comment (mama10) [inline changelog.md:71]: Fixed a bug **where** Material attribute overrides in USD meshes **were** ignored by the importer in some cases.

File: changelog.md [hunk 5]
```diff
@@ -1,3 +1,78 @@
...
+  * inline keyword: Allows to immediately generate a rule and continue working on the resulting shape(s). This means any rule can be used like a builtin shape operation.
+  * union, subtract, intersect keywords: added Boolean 3D operations.
+  * extrude operation: Added support to extrude edges.
+* New operations:
+  * tag operation
+* Changes to existing features:
+  * i operation: The new parameter cleanupMode allows for disabling the numerical geometry cleanup for inserted asset. This improves the support for assets with tiny details.
+  * cleanupGeometry operation: Removing coplanar edges now preserves the first edge.
+  * Floating point precision:
+      * All internal geometry processing was switched form float to double precision. The numerical behaviour is more accurate now. It is possible that some rules change behaviour because e.g a float value comparison gives a different result now.
```
Comment (mama10) [inline changelog.md:25]: e.g.

File: changelog.md [hunk 6]
```diff
@@ -1,3 +1,78 @@
...
+
+# CITYENGINE SDK 3.2.10352 CHANGELOG
+
+This section lists changes compared to CityEngine SDK 3.2.9903.
+
+## General Info
+* The focus of this release is to fix some issues which are important for certain client applications such as ArcGIS Pro. There is no CityEngine version using this version of the SDK.
+
+## PRT API
+* `prt::RuleFileInfo`: fixed wrong rule parameter names (i.e., `prt::RuleFileInfo::Parameter::getName()` and `prt::RuleFileInfo::toXML()`).
```
Comment (mama10) [inline changelog.md:44]: inconsistent capitalization after colon

File: changelog.md [hunk 7]
```diff
@@ -1,3 +1,78 @@
...
+## CGA
+* New operations:
+  * copyUV operation
+* New functions:
+  * geometry.hasUVs function
+
+## Built-In Codecs
+* IFC, DWG Decoders, Encoders:
+  * Fixed a rare deadlock if run in a multithreaded setup. 
+  * Updated to ODA 25.6 library
```
Comment (mama10) [inline changelog.md:58]: inconsistent usage of point at the end

**Review by mama10 — CHANGES_REQUESTED** [review]

File: changelog.md (hunk 1 above)
Comment (matthias-specht) [reply changelog.md:32]: done

File: changelog.md (hunk 2 above)
Comment (matthias-specht) [reply changelog.md:35]: done

File: changelog.md (hunk 3 above)
Comment (matthias-specht) [reply changelog.md:38]: done

File: changelog.md (hunk 4 above)
Comment (matthias-specht) [reply changelog.md:71]: done

File: changelog.md (hunk 5 above)
Comment (matthias-specht) [reply changelog.md:25]: didn't change it, I'd like to keep this as a 1:1 copy of the changelog in the CGA ref

File: changelog.md (hunk 7 above)
Comment (matthias-specht) [reply changelog.md:58]: done

**Comment by matthias-specht** [comment]
thanks for the review! I addressed them and merged to develop. closing this.

**Pull request closed without merging**